### ISSUE TITLE
[Issue]: Cryptic error message when environment variables are not available

### ISSUE THREAD

**Issue by s-h-a-d-o-w** [coreybutler/nvm-windows#1192]
### What happened?

Immediately after installation using the setup exe:

```
C:\Users\andy\dev\alfc>nvm ls

ERROR open \settings.txt: The system cannot find the file specified.

C:\Users\andy\dev\alfc>refreshenv
Refreshing environment variables from registry for cmd.exe. Please wait...Finished..

C:\Users\andy\dev\alfc>nvm ls

    22.11.0
    18.20.5
```

### What did you expect to happen?

An error message that lets me know that nvm can't find the required environment variables.
(Additionally, I think it would be good if the installer asked the user to reboot at the end of the installation to avoid this in the first place. Should probably be optional - for power users who want to use `refreshenv` for the time being.)

### Version

1.1.11 or newer (Default)

### Which version of Windows?

Windows 11+

### Which locale?

English (Default)

### Which shell are you running NVM4W in?

Command Prompt

### User Permissions?

Standard Use, Non-Elevated

### Is Developer Mode enabled?

Yes

### Debug Output

```Shell
Windows Version: 10.0 (Build 22631)

Windows Developer Mode: UNKNOWN (user cannot read registry)

NVM4W Version:      1.1.12
NVM4W Path:         C:\Users\andy\AppData\Roaming\nvm\nvm.exe
NVM4W Settings:     C:\Users\andy\AppData\Roaming\nvm\settings.txt
NVM_HOME:           C:\Users\andy\AppData\Roaming\nvm
NVM_SYMLINK:        C:\Program Files\nodejs
Node Installations: C:\Users\andy\AppData\Roaming\nvm

Total Node.js Versions: 2
Active Node.js Version: none
(run "nvm use <version>" to activate a version)
NVM_SYMLINK does not exist yet. This is auto-created when "nvm use" is run.

IPv6 is enabled. This can slow downloads significantly.

WARNING: The following Node installations are missing npm: v18.20.5
         (Node will still run, but npm will not work on these versions)

PROBLEMS DETECTED
-----------------
"C:\WINDOWS\System32\cmd.exe - nvm  debug" is not an officially supported shell. Some features may not work as expected.        



Find help at https://github.com/coreybutler/nvm-windows/wiki/Common-Issues
```

**Comment by coreybutler** [comment]
Does this still occur in v1.2.1? I added more error handling/messages and added refreshenv to the prompt that opens upon completion of the install.

**Comment by s-h-a-d-o-w** [comment]
I can't find the time to go through the whole uninstall install cycle I'm afraid but if you tested it, it should be fine anyway.
I do hope that you don't rely on `refreshenv` already existing, since it's installed with chocolatey, if I recall correctly.
Also - keep in mind that that only works for that terminal instance. If the user already has e.g. vscode running, all open projects will retain the old environment and would require running refreshenv in each one.

**Comment by coreybutler** [comment]
You're right about refreshenv being part of chocolately. I mistakenly thought it shipped with powershell.

This only needs to work for a single instance of the terminal. The issue is the terminal launches from the installer. Unfortunately, Innosetup doesn't have a proper wait/sleep operation, so a race condition is introduced. Basically, the terminal launches before the environment variables are finished updating.

I implemented a powershell script that waits a moment, then runs refreshenv... but as you pointed out, that cannot be relied on. I'm going to drop the ability to open a prompt from the installer. There are too many issues around this and I don't have enough time to deep-dive into Innosetup.

Thanks for the feedback!

**Comment by s-h-a-d-o-w** [comment]
Glad to hear that it was helpful! 

(And awesome that this is still being maintained! I especially love the quality of life improvements that have brought it more in line with `nvm`, like being able to specify just the major version for commands and whatever is the latest available will be used.)

**Comment by coreybutler** [comment]
We're actually evolving the whole concept to go beyond version management (into environment management), which is where I think the future "quality of life" improvements will come from. For example, the 1.2.x line introduced this:

![Image](https://github.com/user-attachments/assets/1555a81b-c22a-4c13-a2da-d79c5b49cd6b)

I wrote a [recap of the new stuff/direction here](https://opensource.author.io/nvm-for-windows-v120). Version 1.2.3 will be released pretty soon... so if you're waiting to upgrade, that's the version I'd pull the trigger on.

Thanks again!
